shotover-proxy/src/transforms/cassandra/sink_cluster/node.rs

```diff
@@ -18,39 +18,86 @@ pub struct CassandraNode {
 impl CassandraNode {
     pub async fn get_connection(
         &mut self,
-        handshake: &[Message],
-        tls: &Option<TlsConnector>,
-        pushed_messages_tx: &Option<mpsc::UnboundedSender<Messages>>,
+        connection_factory: &ConnectionFactory,
     ) -> Result<&mut CassandraConnection> {
         if self.outbound.is_none() {
             self.outbound = Some(
-                new_connection((self.address, 9042), handshake, tls, pushed_messages_tx).await?,
+                connection_factory
+                    .new_connection((self.address, 9042))
+                    .await?,
             )
         }
 
         Ok(self.outbound.as_mut().unwrap())
     }
 }
 
-pub async fn new_connection<A: ToSocketAddrs>(
-    address: A,
-    handshake: &[Message],
-    tls: &Option<TlsConnector>,
-    pushed_messages_tx: &Option<mpsc::UnboundedSender<Messages>>,
-) -> Result<CassandraConnection> {
-    let outbound = CassandraConnection::new(
-        address,
-        CassandraCodec::new(),
-        tls.clone(),
-        pushed_messages_tx.clone(),
-    )
-    .await?;
-
-    for handshake_message in handshake {
-        let (return_chan_tx, return_chan_rx) = oneshot::channel();
-        outbound.send(handshake_message.clone(), return_chan_tx)?;
-        return_chan_rx.await?;
+#[derive(Clone, Debug)]
+pub struct ConnectionFactory {
+    init_handshake: Vec<Message>,
+    use_message: Option<Message>,
+    tls: Option<TlsConnector>,
+    pushed_messages_tx: Option<mpsc::UnboundedSender<Messages>>,
+}
+
+impl ConnectionFactory {
+    pub fn new(tls: Option<TlsConnector>) -> Self {
+        Self {
+            init_handshake: vec![],
+            use_message: None,
+            tls,
+            pushed_messages_tx: None,
+        }
+    }
+
+    pub fn clone_config(&self) -> Self {
```

Comment:
its a bit concerning to me having multiple clone implementations and trying to remember which is which.
Maybe we could call this `new_with_same_tls` or `new_with_same_config` so it doesnt feel like we are cloning?

Reply:
Oh we could completely avoid needing the second clone implementation by moving tls outside of this struct and back int o CassandraSinkCluster.
To me that separation makes sense because tls is immutable config while the rest of this struct is mutable state.

Up to you.

Reply:
We need to be able to clone it to pass to the topology task. In that sense it's immutable, once it's given to the task it is no longer mutated.

Maybe I should add more docs but to me it was clear that this clone is for when you only want the config for the new connection (just TLS for now), and the default clone is for when you want everything within the struct.